internal/updatecheck/update.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2020 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package updatecheck
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"path/filepath"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/pkg/errors"
+	"k8s.io/klog"
+
+	"sigs.k8s.io/krew/internal/version"
+)
+
+const (
+	githubVersionURL = "https://api.github.com/repos/kubernetes-sigs/krew/releases/latest"
+
+	upgradeNotification = `A newer version of krew is available (%s -> %s). 
+Run "kubectl krew upgrade" to get the newest version!`
+)
+
+// for testing
+var versionURL = githubVersionURL
+
+// CheckVersion returns a notification message to inform the user
+// about a new version of krew, or an empty string. The notification
+// will only be emitted once a day.
+func CheckVersion(basePath string) (string, error) {
+	f := filepath.Join(basePath, "last_update_check")
+	if lastCheck := loadTimestamp(f); time.Since(lastCheck).Hours() <= 24 {
```

Comment:
could you put the range to the top as constant?

an idea might be introducing another environment variable to customize it.

Reply:
Just hardcoded const is good